openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -264,6 +271,35 @@ def certificate_data(self):
         if self.enrollment_object:
             return get_cert_data(self.effective_user, self.course, self.enrollment_object.mode)
 
+    @property
+    def entrance_exam_data(self):
+        return {
+            'entrance_exam_current_score': get_entrance_exam_score_ratio(
+                self.course_grade, get_entrance_exam_usage_key(self.overview),
+            ),
+            'entrance_exam_enabled': course_has_entrance_exam(self.overview) and self.overview.entrance_exam_enabled,
+            'entrance_exam_id': self.overview.entrance_exam_id,
+            'entrance_exam_minimum_score_pct': self.overview.entrance_exam_minimum_score_pct,
+            'entrance_exam_passed': user_has_passed_entrance_exam(self.effective_user, self.overview),
+        }
+        # if course_has_entrance_exam(self.overview):
+        #     return {
+        #         'entrance_exam_current_score': get_entrance_exam_score_ratio(
+        #             self.course_grade, get_entrance_exam_usage_key(self.overview),
+        #         ),
+        #         'entrance_exam_enabled': self.overview.entrance_exam_enabled,
+        #         'entrance_exam_id': self.overview.entrance_exam_id,
+        #         'entrance_exam_minimum_score_pct': self.overview.entrance_exam_minimum_score_pct,
+        #         'entrance_exam_passed': user_has_passed_entrance_exam(self.effective_user, self.overview),
+        #     }
+        # return {
+        #     'entrance_exam_current_score': 0.0,
+        #     'entrance_exam_enabled': False,
+        #     'entrance_exam_id': '',
+        #     'entrance_exam_minimum_score_pct': self.overview.entrance_exam_minimum_score_pct,
+        #     'entrance_exam_passed': False,
+        # }
```

Comment:
Happy to hear if there is a preference between these two, otherwise I think I'll opt for the above version.

Reply:
The top is fine.